src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp

```diff
@@ -235,37 +235,37 @@ bool Transformations::fuse_type_to_convert(const std::shared_ptr<ov::Node>& node
 
-    // For Convert node, converting precision from floating point to boolean will lead to mathematical
-    // error, because here the output precision boolean is replaced by u8:
-    //  - floating point value 0.01 is converted to be 1 for boolean, but 0 for u8 - need to insert Ceil.
-    //  - floating point value 256 is converted to be 1 for boolean, but 0 for u8 - need to insert Min(x, UINT8_MAX)
-    //  - floating point value -256 is converted to be 1 for boolean, but 0 for u8 - need to insert Abs before Min.
-    // Thus an Abs, Ceil and Min nodes should be added before the Convert node for this scenario.
-    if (convert->input(0).get_element_type().is_real() &&
-        convert->get_convert_element_type() == ov::element::boolean && to.is_integral_number()) {
+    if (convert->get_convert_element_type() == ov::element::boolean && to.is_integral_number()) {
+        // For Convert node, converting precision from numerical data types to boolean will lead to mathematical
+        // error, because here the output precision boolean is replaced by u8:
+        //  - floating point value 0.01 is converted to be 1 for boolean, but 0 for u8 - need to insert Ceil.
+        //  - either float or int values should be clipped with the interval [0; 1] to mimic bool cast behavior, i.e.
+        //  0 - is false, 1 - is true
+        //  - to perform clamping correctly an Abs op should be inserted before Clamp
+        // Thus an Abs, Ceil and Clamp nodes should be added before the Convert node for this scenario.
         ov::pass::NodeRegistry reg;
         const auto& in_prec = convert->get_input_element_type(0);
-        auto data = convert->input_value(0).get_node_shared_ptr();
+        auto parent_node = convert->input_value(0).get_node_shared_ptr();
         auto item = precisions.find(in_prec);
         if (item != precisions.end()) {
-            // Add convert node for unsupported precision, such as FP64
-            data = reg.make<ov::opset10::Convert>(data, item->second);
+            // Add convert node for unsupported precision, such as FP64 or INT64
+            parent_node = reg.make<ov::opset10::Convert>(parent_node, item->second);
         }
-        const auto abs = reg.make<ov::opset10::Abs>(data);
-        const auto to_max_value = reg.make<ov::opset10::Constant>(ov::util::make_tensor_of_max_value(to));
-        const auto to_max_convert = reg.make<ov::opset10::Convert>(to_max_value, abs->get_output_element_type(0));
-        const auto min = reg.make<ov::opset10::Minimum>(abs, to_max_convert);
-        const auto ceil = reg.make<ov::opset10::Ceiling>(min);
-        const auto new_convert = reg.make<ov::opset10::Convert>(ceil, to);
+        if (in_prec.is_signed()) {
+            parent_node = reg.make<ov::opset10::Abs>(parent_node);
+        }
+        if (in_prec.is_real()) {
+            parent_node = reg.make<ov::opset10::Ceiling>(parent_node);
+        }
+        parent_node = reg.make<ov::opset10::Clamp>(parent_node, 0, 1);
```

Comment:
Do we really replace `Min(x, MAX)` with `Clamp`? I think we can skip execution of `Max(x, 0)` in `Clamp` and evaluate only `Max(x, 1)`. Please see:
If there is unsigned number, it will be in `[0,....]`. Minimum value is already `0`.
If there is signed number, firstly we execute `Abs` on 255 line and will get unsigned value again which is in interval `[0,...]`. The possible op `Ceiling` doesn't affect this interval.
Thus, the op `Clamp` always get value `x` in interval `[0,...]`

Reply:
First, please refer to the Clamp formula `clamp(x) = min(max(x, min_value), max_value), so the both operations min/max are performed. Moreover, here we interpret an interval [-inf, inf] as boolean values, where only exact 0 is false (i.e. 0), and all the other values are true (i.e. 1).
Abs map the [-inf, inf] interval to [0, inf].
Ceiling makes it desecrate, mapping only exact 0 to 0, all the other numbers are in the integer interval [1, inf].
Clamp simply cut off all the values >1, replacing them with 1, of course preserving exact zeroes.
Is it clear now? Is something missed in the algorithm?

Reply:
Actually we can replace Clamp with only the Min operation, as due to the Abs, the whole interval is greater or equal zero, so we don't need to perform max(x, 0). Thank you for the suggestion.

Reply:
Agreed to leave Clamp as it has only one input while Min has two.